crates/core_simd/src/elements/int.rs

```diff
@@ -191,6 +191,25 @@ pub trait SimdInt: Copy + Sealed {
 
     /// Returns the cumulative bitwise "xor" across the lanes of the vector.
     fn reduce_xor(self) -> Self::Scalar;
+
+    /// Reverses the byte order of each element.
+    fn swap_bytes(self) -> Self;
+
+    /// Reverses the order of bits in each elemnent.
+    /// The least significant bit becomes the most significant bit, second least-significant bit becomes second most-significant bit, etc.
+    fn reverse_bits(self) -> Self;
+
+    /// Returns the number of leading zeros in the binary representation of each element.
+    fn leading_zeros(self) -> Self;
```

Comment:
Do we think it makes sense for these functions to return `Self` (a signed int) or should it have a more complex signature and return the equivalent unsigned type?

Reply:
which exact functions are you referring to? for `leading_zeros`, any integer type with >= 3 bits will never wrap no matter which signedness, so we have no problem until we get generic-integer element types (if that ever happens...). I think `reverse_bits` should return `Self` for consistency with `i32::reverse_bits`

Reply:
for `leading_zeros` it might be nice to return the corresponding unsigned type with the same bit-width, since that's consistent with the integer types, and also since that works for all bit lengths including 0, 1, and 2-bits.

Reply:
I just mean the counting functions, like `leading_zeros`.  I agree that `Self` will always fit the result, I meant more if it's inconsistent (with scalars) to return a signed type, instead of the unsigned int with the same size.  This is definitely the simpler option

Reply:
ah, ok. well:
> for leading_zeros it might be nice to return the corresponding unsigned type with the same bit-width, since that's consistent with the integer types, and also since that works for all bit lengths including 0, 1, and 2-bits.

plus the bit-counting functions are imo better thought of as unsigned rather than signed integer operations